docs/concepts/controllers/statefulsets.md

```diff
@@ -0,0 +1,163 @@
+---
+---
+
+{% capture overview %}
+**Stateful Sets are a beta feature in 1.5. This feature replaces the deprecated 
+Pet Sets feature from 1.4. Users of Pet Sets are referred to the 1.5 
+[Upgrade Guide](/docs/tasks/stateful-set/upgrade-from-petsets-to-stateful-sets/)
+for further information on how to upgrade existing Pet Sets to Stateful Sets.**
+
+A Stateful Set is a Controller that ensures that provides a unique identity to 
+its Pods, and provides garuantees about the ordering of deployment and scaling.
+{% endcapture %}
+
+{% capture body %}
+
+### When to Use a Stateful Set
+Stateful Sets are valuable for applications that require one or more of the 
+following.
+
+* Stable, unique network identifiers.
+* Stable, persistent storage.
+* Ordered, graceful deployment and scaling.
+* Ordered, graceful deletion and termination.
+
+As it is generally easier to manage, if an application doesn't require any of 
+the above guarantees, and if it is feasible to do so, it should be deployed as 
+a set of stateless replicas.
+
+### Limitations
+* Stateful Set is a beta resource, not available in any Kubernetes release prior to 1.5.
+* As with all alpha/beta resources, it can be disabled through the `--runtime-config` option passed to the apiserver.
+* The only mutable field on a Stateful Set is `replicas`
```

Comment:
We can change the PodTemplateSpec IIRC. Perhaps we should remove this assertion.